libraries/AP_RangeFinder/AP_RangeFinder_BLPing.cpp

```diff
@@ -66,12 +36,65 @@ void AP_RangeFinder_BLPing::update(void)
 
 void AP_RangeFinder_BLPing::init_sensor()
 {
-    // request distance from sensor
-    send_message(BLDPIN_MSGID_DISTANCE_SIMPLE, nullptr, 0);
+    // Set message interval between pings in ms
+    uint16_t ping_interval = _sensor_rate_ms;
+    protocol.send_message(uart, PingProtocol::MessageId::SET_PING_INTERVAL, reinterpret_cast<uint8_t*>(&ping_interval), sizeof(ping_interval));
+
+    // Send a message requesting a continuous
+    uint16_t continuous_message = static_cast<uint16_t>(PingProtocol::MessageId::DISTANCE_SIMPLE);
```

Comment:
If this fails, we wait for the 1 second timeout to expire and try again. Does the sensor send an ack or anything? I think this function should return bool, but that's for another patch.

Reply:
This will add extra logic that is not inside the scope of this patch.
The patch only breaks the parse logic in a different class and uses the continuous request.
We can improve the parse class to deal with the entire ping-protocol standard in future PRs.